sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -661,6 +663,69 @@ public CosmosAsyncConflict getConflict(String id) {
                    }).map(offerResourceResponse -> offerResourceResponse.getResource().getThroughput());
     }
 
+    /**
+     * Replace the throughput .
+     *
+     * @param throughputProperties the throughput properties
+     * @return the mono containing throughput response
+     */
+    public Mono<ThroughputResponse> replaceThroughput(ThroughputProperties throughputProperties) {
+        return this.read()
+                   .flatMap(response -> this.database.getDocClientWrapper()
+                                            .queryOffers("select * from c where c.offerResourceId = '"
+                                                             + response.getProperties()
+                                                                   .getResourceId()
```

Comment:
I would use parametrized SqlQuerySpec instead of concating strings

Reply:
Changed to use querySpec